File: app/src/main/java/mb/pokequiz/ui/BaseActivity.java
package mb.pokequiz.ui;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

import javax.inject.Inject;

import butterknife.ButterKnife;
import mb.pokequiz.PokeApplication;
import mb.pokequiz.data.api.PokeService;
import rx.Subscription;
import rx.subscriptions.CompositeSubscription;

/**
 * Created by mbpeele on 2/1/16.
 */
public class BaseActivity extends AppCompatActivity {

    @Inject PokeService pokeService;

    private CompositeSubscription compositeSubscription;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        ((PokeApplication) getApplication()).getApplicationComponent().inject(this);
        compositeSubscription = new CompositeSubscription();
    }

    @Override
    public void setContentView(int layoutResID) {
        super.setContentView(layoutResID);
        ButterKnife.bind(this);
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        compositeSubscription.unsubscribe();
        ButterKnife.unbind(this);
    }

    public void addSubscription(Subscription subscription) {
        compositeSubscription.add(subscription);
    }

    public void removeSubscription(Subscription subscription) {
        compositeSubscription.remove(subscription);
    }

    public void unsubscribeFromAllSubscriptions() {
        compositeSubscription.unsubscribe();
    }
}

File: app/src/main/java/mb/pokequiz/ui/base/BaseActivity.kt
package mb.pokequiz.ui.base

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import mb.pokequiz.PokeApplication
import mb.pokequiz.dagger.activity.ActivityComponent
import mb.pokequiz.dagger.activity.ActivityModule
import mb.pokequiz.dagger.activity.DaggerActivityComponent

open class BaseActivity : AppCompatActivity() {

    lateinit var component : ActivityComponent

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        component = DaggerActivityComponent.builder()
                .appComponent(PokeApplication.component)
                .activityModule(ActivityModule())
                .build()
    }

}
